server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -48,30 +27,56 @@ class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit])
             case Some(response) => response
             case None           => ServerResponse.notFound
           }
-          .map(toHttpResponse(_, request))
-          .map(flushResponse(ctx, request, _))
+          .map((serverResponse: ServerResponse[NettyResponse]) => {
+            serverResponse.body.handle(
+              byteBufHandler = (byteBuf) => {
+                val res = new DefaultFullHttpResponse(req.protocolVersion(), HttpResponseStatus.valueOf(serverResponse.code.code), byteBuf)
+
+                res.setHeadersFrom(serverResponse)
+                res.handleContentLengthHeader(byteBuf.readableBytes())
+                res.handleCloseAndKeepAliveHeaders(request)
+
+                ctx.writeAndFlush(res).closeIfNeeded(req)
+              },
+              chunkedInputHandler = (httpChunkedInput, length) => {
+                val resHeader: DefaultHttpResponse =
+                  new DefaultHttpResponse(req.protocolVersion(), HttpResponseStatus.valueOf(serverResponse.code.code))
+
+                resHeader.setHeadersFrom(serverResponse)
+                resHeader.handleContentLengthHeader(length)
+                resHeader.handleCloseAndKeepAliveHeaders(request)
+
+                ctx.write(resHeader)
+                ctx.writeAndFlush(httpChunkedInput, ctx.newProgressivePromise()).closeIfNeeded(req)
```

Comment:
maybe we can create this "progressive promise" when creating the body and then use it in the listener?

Reply:
Unfortunately not, as far as I understand `ProgresivePromise` can be created only with `ChannelHandlerContext` presence.